### DISCUSSION TITLE
How can I test the performance of k3s? Can you share the testing tools and usage methods?

### DISCUSSION THREAD

**Discussion by hanzhc** [k3s-io/k3s#10482] [Q&A]
I have set up a k3s cluster. How can I perform density testing and load testing on it?

**Comment by brandond** [comment]
Define your parameters. What do you want to test? Network throughput? Disk IO? Number of pods you can run? Number of custom resources you can manage? Kubernetes casts a wide net, there is no one tool that will give you universal performance figures for all of its components. Any tool that works on other Kubernetes distros should also work on K3s.

You can see some information on how we collected our baseline figures at https://docs.k3s.io/reference/resource-profiling - for anything more advanced than that, you should not be looking at k3s-specific tools.
